lib/elixir/src/elixir_clauses.erl

```diff
@@ -232,25 +232,28 @@ assert_at_most_once(Kind, [_ | Rest], Count, Fun) ->
   assert_at_most_once(Kind, Rest, Count, Fun).
 
 format_error({bad_or_missing_clauses, {Kind, Key}}) ->
-  io_lib:format("expected -> clauses for ~ts in ~ts", [Key, Kind]);
+  io_lib:format("expected -> clauses for :~ts in \"~ts\"", [Key, Kind]);
 format_error({bad_or_missing_clauses, Kind}) ->
-  io_lib:format("expected -> clauses in ~ts", [Kind]);
-format_error({missing_do, Kind}) ->
-  io_lib:format("missing do keyword in ~ts", [Kind]);
-format_error({invalid_args, Kind}) ->
-  io_lib:format("invalid arguments for ~ts", [Kind]);
+  io_lib:format("expected -> clauses in \"~ts\"", [Kind]);
+
 format_error({duplicated_clauses, Kind, Key}) ->
-  io_lib:format("duplicated ~ts clauses given for ~ts", [Key, Kind]);
-format_error({unexpected_keyword, Kind, Keyword}) ->
-  io_lib:format("unexpected keyword ~ts in ~ts", [Keyword, Kind]);
+  io_lib:format("duplicated :~ts clauses given for \"~ts\"", [Key, Kind]);
+
+format_error({unexpected_option, Kind, Option}) ->
+  io_lib:format("unexpected option ~ts in \"~ts\"", ['Elixir.Macro':to_string(Option), Kind]);
+
 format_error({wrong_number_of_args_for_clause, Expected, Kind, Key}) ->
-  io_lib:format("expected ~ts for ~ts clauses (->) in ~ts", [Expected, Key, Kind]);
+  io_lib:format("expected ~ts for :~ts clauses (->) in \"~ts\"", [Expected, Key, Kind]);
+
 format_error(multiple_after_clauses_in_receive) ->
-  "expected a single -> clause for after in receive";
+  "expected a single -> clause for :after in \"receive\"";
+
 format_error(missing_do_or_after_in_receive) ->
-  "missing do or after keyword in receive";
+  "missing :do or :after in \"receive\"";
+
 format_error(missing_keyword_in_try) ->
```

Comment:
Great catch. Was thinking, what if we unify this error as well and have something like

```erlang
format_error({missing_options, Kind, Opts}) ->
  StringOpts = 'Elixir.Enum':map_join(Opts, <<"/">>, fun 'Elixir.Macro':to_string/1),
  io_lib:format("missing ~ts in \"~ts\"", [StringOpts, Kind]);
```

Wdyt?